### PR TITLE
Call cl::ResetAllOptionOccurrences() before reparsing LLVM options

### PR DESCRIPTION
**Pull request by brianwatling**
This change makes ICompiler::Create check whether options have
previously been parsed, regardless of whether any instances are
currently active, and calls cl::ResetAllOptionOccurrences() when options
are changing. The error handler is also removed when the last compiler
instance is destroyed. These changes fix crashes while running Vulkan
CTS.

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
Can one of the admins verify this patch?

**Comment by JacobHeAMD** [comment]
ok to test

**Comment by JacobHeAMD** [comment]
test this please

**Comment by JacobHeAMD** [comment]
The shaderDB test is failed, please check the details of "LLPC Docker CI".

File: llpc/context/llpcCompiler.cpp [hunk 1]
```diff
@@ -177,6 +177,7 @@ enum ShaderReplaceMode {
 };
 
 static ManagedStatic<sys::Mutex> SCompilerMutex;
```
Comment (amdrexu) [inline llpc/context/llpcCompiler.cpp:179]: Help to do this change: SCompilerMutex -> CompilerMutex, SOptionHash -> OptionHash
I think this is caused by renaming of coding style change

Comment (brianwatling) [reply llpc/context/llpcCompiler.cpp:179]: Thanks for the suggestion. I had some lines deleted that I hand't intended to remove - fixed now

**Comment by amdvlk-admin** [comment]
### Test summary for commit db92a2e28f13c4d1cf853d747c7503f7c98ceabf
<details open><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master b601c88aeca7a7b08becb3d32709de383c8ee428</li>
<li>METROHASH: amd-master 2b6fee002db6cc92345b02aeee963ebaaf4c0e2f</li>
<li>PAL: dev b5e6dde2a0aa5dd2ce6d794e58d89ef715570f2d</li>
<li>SPVGEN: dev e843618d71c8263030551e7692b7683f20f627fe</li>
<li>XGL: dev 2c44bbc7b58efba12c501f71f5b268f83c1bdad5</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 5ae05ed8bd4f232c8580db96c021cad1e5c300e3</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 955/955 (100.0%)</li>
<li>Failed: 0/955 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.2.0.2-fetch-sources-fix)</summary><ul>


<details open><summary>Ubuntu 18.04, Ellesmere</summary><ul>

<li>Passed: 103142/103142 (100.0%)</li>
<li>Failed: 0/103142 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Navi10</summary><ul>

<li>Passed: 2198/2199 (100.0%)</li>
<li><details><summary>Failed: 1/2199 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.api.object_management.multithreaded_shared_resources.graphics_pipeline
Stack trace: result: VK_ERROR_OUT_OF_HOST_MEMORY at vktApiObjectManagementTests.cpp:1827
```
</p></details></li>


</ul></details>

<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 103198/103198 (100.0%)</li>
<li>Failed: 0/103198 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit c91888fc668e6deeb43c04d42ab55d2fb3a8daea
<details open><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master b601c88aeca7a7b08becb3d32709de383c8ee428</li>
<li>METROHASH: amd-master 2b6fee002db6cc92345b02aeee963ebaaf4c0e2f</li>
<li>PAL: dev e2f8eb0ffe21751b300995e795965d37e8d6dd67</li>
<li>SPVGEN: dev e843618d71c8263030551e7692b7683f20f627fe</li>
<li>XGL: dev 2c44bbc7b58efba12c501f71f5b268f83c1bdad5</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev f02738a5cf6c158c8c5b25a5048951f3059cabe2</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 958/958 (100.0%)</li>
<li>Failed: 0/958 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.2.0.2-fetch-sources-fix)</summary><ul>


<details open><summary>Ubuntu 18.04, Ellesmere</summary><ul>

<li>Passed: 103142/103142 (100.0%)</li>
<li>Failed: 0/103142 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Navi10</summary><ul>

<li>Passed: 103198/103198 (100.0%)</li>
<li>Failed: 0/103198 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 103198/103198 (100.0%)</li>
<li>Failed: 0/103198 (0.0%)</li>
</ul></details>


</ul></details>

**Pull request merged**